tests/rest/datacube/test_datacube.py

```diff
@@ -136,6 +136,19 @@ def test_load_collection_connectionless_temporal_extent_shortcut(self):
             }
         }
 
+    def test_load_collection_connectionless_shapely_spatial_extent(self):
+        polygon = shapely.Polygon(((0.0,1.0),(2.0,1.0),(3.0,2.0),(1.5,0.0),(0.0,1.0)))
+        cube = DataCube.load_collection("T3", spatial_extent=polygon)
+        assert cube.flat_graph() == {
+            "loadcollection1": {
+                "arguments": {"id": "T3", "spatial_extent":
+                    {'coordinates': (((0.0,1.0),(2.0,1.0),(3.0,2.0),(1.5,0.0),(0.0,1.0)),),'type': 'Polygon'},
+                     "temporal_extent": None},
+                "process_id": "load_collection",
+                "result": True,
+            }
+        }
+
```

Comment:
_get_geometry_argument also adds support for loading geometry from a local file, which should be covered by tests.
e.g. for inspiration see https://github.com/Open-EO/openeo-python-client/blob/e106045dfa3c05821c5c3e9d5abd4cd3d7b6514b/tests/rest/datacube/test_datacube100.py#L689-L692